Tests/DeckOfPlayingCardsTests/DeckTest.swift

```diff
@@ -1,62 +1,70 @@
-/*
- This source file is part of the Swift.org open source project
- 
- Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
- Licensed under Apache License v2.0 with Runtime Library Exception
- 
- See http://swift.org/LICENSE.txt for license information
- See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
-*/
-
-import XCTest
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the example package deckofplayingcard open source project
+//
+// Copyright (c) 2015-2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
 
-import PlayingCard
 import DeckOfPlayingCards
+import PlayingCard
+import Testing
 
-class DeckTests: XCTestCase {
-    func testStandard52CardDeck() {
-        var countByPlayingCard: [PlayingCard: Int] = [:]
+struct DeckTests {
+  @Test
+  func standard52CardDeck() {
+    var countByPlayingCard: [PlayingCard: Int] = [:]
 
-        var deck = Deck.standard52CardDeck()
-        while let playingCard = deck.deal() {
-            countByPlayingCard[playingCard, default: 0] += 1
-        }
+    var deck = Deck.standard52CardDeck()
+    while let playingCard = deck.deal() {
+      countByPlayingCard[playingCard, default: 0] += 1
+    }
 
-        XCTAssertEqual(countByPlayingCard.count, 52)
-        XCTAssertTrue(countByPlayingCard.values.allSatisfy { $0 == 1 })
+    #expect(countByPlayingCard.count == 52)
+    #expect(countByPlayingCard.values.allSatisfy({ $0 == 1 }))
 
-        for rank in Rank.allCases {
-            for suit in Suit.allCases {
-                let playingCard = PlayingCard(rank: rank, suit: suit)
-                XCTAssertEqual(countByPlayingCard[playingCard], 1)
-            }
-        }
-    }
-    
-    func testDeal() {
-        let playingCard = PlayingCard(rank: .ace, suit: .clubs)
-        var deck: Deck = [playingCard]
-        
-        XCTAssertEqual(deck.deal(), playingCard)
-        XCTAssertNil(deck.deal())
-    }
-    
-    func testCountEmptyDeckHasZeroCards() {
-        let deck = Deck()
-        XCTAssertEqual(deck.count, 0)
-    }
-    
-    func testCountStandard52CardDeckHas52Cards() {
-        let deck = Deck.standard52CardDeck()
-        
-        XCTAssertEqual(deck.count, 52)
-    }
-    
-    func testCountDealingDecreasesCountByOne() {
-        var deck = Deck([PlayingCard(rank: .ace, suit: .spades), PlayingCard(rank: .queen, suit: .hearts)])
-        
-        XCTAssertEqual(deck.count, 2)
-        XCTAssertNotNil(deck.deal())
-        XCTAssertEqual(deck.count, 1)
+    for rank in Rank.allCases {
```

Comment:
Parameterized testing opportunity here?

Reply:
Ooh - good idea, thank you. I'll update.

Reply:
Er, hurm. This doesn't look like a great option - the code bit there is verifying that 1 of each card was dealt out, so there's not an easy set of parameters to hand in prior to validate